backend/src/schema/resolvers/reports.spec.js

```diff
@@ -1,37 +1,73 @@
 import { GraphQLClient } from 'graphql-request'
 import Factory from '../../seed/factories'
-import { host, login } from '../../jest/helpers'
-import { neode } from '../../bootstrap/neo4j'
+import { host, login, gql } from '../../jest/helpers'
+import { getDriver, neode } from '../../bootstrap/neo4j'
+import { createTestClient } from 'apollo-server-testing'
+import createServer from '../.././server'
 
 const factory = Factory()
 const instance = neode()
+const driver = getDriver()
 
-describe('report', () => {
+describe('report mutation', () => {
   let reportMutation
   let headers
-  let returnedObject
+  let client
   let variables
   let createPostVariables
   let user
   const categoryIds = ['cat9']
 
+  const action = () => {
+    reportMutation = gql`
+      mutation($resourceId: ID!, $reasonCategory: ReasonCategory!, $reasonDescription: String!) {
+        report(
+          resourceId: $resourceId
+          reasonCategory: $reasonCategory
+          reasonDescription: $reasonDescription
+        ) {
+          createdAt
+          reasonCategory
+          reasonDescription
+          type
+          submitter {
+            email
+          }
+          user {
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-10-14T08:13:01Z" title="Monday, October 14th 2019, 10:13:01 am +02:00">Oct 14, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -1,37 +1,55 @@
 import { GraphQLClient } from 'graphql-request'
 import Factory from '../../seed/factories'
-import { host, login } from '../../jest/helpers'
-import { neode } from '../../bootstrap/neo4j'
+import { host, login, gql } from '../../jest/helpers'
+import { getDriver, neode } from '../../bootstrap/neo4j'
+import { createTestClient } from 'apollo-server-testing'
+import createServer from '../.././server'
 
 const factory = Factory()
 const instance = neode()
+const driver = getDriver()
 
-describe('report', () => {
+describe('report mutation', () => {
   let reportMutation
   let headers
+  let client
   let returnedObject
   let variables
   let createPostVariables
   let user
   const categoryIds = ['cat9']
 
+  const action = () => {
+    // because of the template `${returnedObject}` the 'gql' tag from 'jest/helpers' is not working here
+    reportMutation = `
+      mutation($resourceId: ID!, $reasonCategory: String!, $reasonDescription: String!) {
+        report(resourceId: $resourceId, reasonCategory: $reasonCategory, reasonDescription: $reasonDescription) ${returnedObject}
+      }
+    `
+    client = new GraphQLClient(host, {
+      headers,
+    })
+    return client.request(reportMutation, variables)
+  }
+
   beforeEach(async () => {
-    returnedObject = '{ id }'
+    returnedObject = '{ createdAt }'
```


can we please refactor this old code to use `toMatchObject`, so that we can have one constant with multiple attributes requested back and in each it block check for what it is we want to check?

I can do it, if you'd like

Reply:
> [<img alt="Tirokk" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2019-10-14T15:24:54Z" title="Monday, October 14th 2019, 5:24:54 pm +02:00">Oct 14, 2019</time>_

---


I have done locally …
Very good hint @mattwr18 !!! 😎